lib/cocoapods/bazel/target.rb

```diff
@@ -374,8 +374,10 @@ def to_rule_kwargs
         kwargs[:runtime_deps] = []
         kwargs[:sdk_dylibs] = file_accessors.flat_map { |fa| fa.spec_consumer.libraries }.sort.uniq
         kwargs[:sdk_frameworks] = file_accessors.flat_map { |fa| fa.spec_consumer.frameworks }.sort.uniq
+        kwargs[:testonly] = true if kwargs[:sdk_frameworks].include? 'XCTest'
```

Comment:
@dierksen is this for the explict PCM stuff? I think the issue there is we need to add framework PCMS based on `sdk_frameworks`  https://github.com/bazel-ios/rules_ios/pull/562#discussion_r1022122823

There are other reasons we want to do this - but it also prevents having to do something like this

Reply:
no, this is to account for changes in https://github.com/bazelbuild/rules_swift/pull/868 that are preventing us from bumping `rules_ios`.

Reply:
Well it looks like that PR is going to force you to use `test_only` all of the place 🤔 

Maybe the same issue applies: e.g. adding the paths more tactfully. I know one sticking point here is wanting to use `XCTest` in other contexts

Reply:
`wanting to use XCTest in other contexts` inside of binaries that ship with the app you mean? I'm not sure I see the harm in the change here, but I might not be understanding something.

Reply:
FWIW we had to split a few modules into two halves: one for mocks/fakes that were used in test apps and one for assertion utilities that were used in unit tests and thus got `testonly` to import XCTest

Reply:
>  I'm not sure I see the harm in the change here,

➕  I'm curious to know as well. Also how this overlaps with cocoapods/Xcode defaults. One thing I thought of was when I end up using `XCTest` framework elsewhere than a test - or you have tests that don't use `XCTest` ..

Reply:
The rules_swift change made it so any target that wishes to link XCTest must be marked testonly. 

This change correctly applies the testonly attribute to the BUILD files in order to work with the new rules_swift changes.

Reply:
Cool - makes sense to me: re - making it work with latest rules_swift.

I was thinking if it'd be worth to improve on that - e.g. if there is another way: e.g. auto-detecting adding the search path when you use `XCTest` VS adding other hinges no `testonly`.. 

I personally wouldn't bind the 2 together. Purity aside today when a developer adds `XCTest` to some `apple_framework` in rules_ios today it kind of just works - but now we'll expect them to fix these new errors by having to add `testonly`.. 

That said, you can probably preserve the old behavior by just injecting `testonly` in rules_ios VS doing analysis phase detection of it.